lib/type_check.ex

```diff
@@ -322,4 +323,99 @@ defmodule TypeCheck do
       {:error, exception} -> raise exception
     end
   end
+  
+  @doc """
+  Ensure at runtime that arguments at result of function call conform the function spec.
+  
+  The function spec is extracted at compile time from the regular Elixir (oe Erlang) `@spec`.
+
+  ## Experimental
+  
+  This macro is experimental. Use it at your own risk only in a test-covered code.
+  If it explodes, please, [open an issue](https://github.com/Qqwy/elixir-type_check/issues).
+  
+  ## Examples
+  
+      iex> TypeCheck.ensure_types!(Kernel.abs(-13))
+      13
+      iex> TypeCheck.ensure_types!(Kernel.abs("hi"))
+      ** (TypeCheck.TypeError) At lib/type_check.ex:280:
+          `"hi"` is not a number.
+  """
+  @spec ensure_types!(Macro.t) :: Macro.t
+  defmacro ensure_types!(expr) do
+    {module, function, args} = Parser.ast_to_mfa(expr)
+    {:ok, spec} = Parser.fetch_spec(module, function, length(args))
```

Comment:
On `{:error, problem}` being returned from `fetch_spec` you might want to raise a `TypeCheck.CompileError, problem`.